cocos/2d/CCTextFieldTTF.cpp

```diff
@@ -679,29 +681,24 @@ const std::string& TextFieldTTF::getPlaceHolder() const
 
 void TextFieldTTF::setCursorEnabled(bool enabled)
 {
-    if (_currentLabelType == LabelType::TTF)
-    {
-        if (_cursorEnabled != enabled)
-        {
-            _cursorEnabled = enabled;
-            if (_cursorEnabled)
-            {
-                _cursorPosition = _charCount;
-
-                scheduleUpdate();
-            }
-            else
-            {
-                _cursorPosition = 0;
-
-                unscheduleUpdate();
-            }
-        }
-    }
-    else
-    {
-        CCLOG("TextFieldTTF cursor worked only LabelType::TTF");
-    }
+	if (_cursorEnabled != enabled)
+	{
+	    _cursorEnabled = enabled;
+	    if (_cursorEnabled)
+	    {
+	        _cursorPosition = _charCount;
+	
+			if (_currentLabelType == LabelType::TTF)
```

Comment:
format error, a `Tab` should be replaced by 4 `Space`, and can you add a test case to show the fix? I noticed that the old test didn't test cursor. 

> the content needed to add test case, you can refer to [pull/18552/files](https://github.com/cocos2d/cocos2d-x/pull/18552/files)

Reply:
Ok, thank you, I missed that. Test implementations will take some time.